### PR TITLE
feat: add 'gno env' subcommand

### PR DESCRIPTION
**Pull request by gfanton**
ref: #1217 

<details><summary>Contributors' checklist...</summary>

- [ ] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

File: gno.land/pkg/gnoland/app.go [hunk 1]
```diff
@@ -2,12 +2,11 @@ package gnoland
 
 import (
 	"fmt"
-	"os"
-	"os/exec"
 	"path/filepath"
 	"strings"
 
 	"github.com/gnolang/gno/gno.land/pkg/sdk/vm"
+	"github.com/gnolang/gno/gnovm/pkg/gnoroot"
```
Comment (moul) [inline gno.land/pkg/gnoland/app.go:9]:
```suggestion
	"github.com/gnolang/gno/gno.land/pkg/gnoroot"
```

File: tm2/pkg/crypto/keys/client/common.go [hunk 2]
```diff
@@ -25,11 +25,19 @@ var DefaultBaseOptions = BaseOptions{
 func HomeDir() string {
 	// if environment variable is set, always use that.
 	// otherwise, use config dir (varies depending on OS) + "gno"
-	var err error
-	dir := os.Getenv("GNO_HOME")
+
+	dir := os.Getenv("GNOHOME")
```
Comment (moul) [inline tm2/pkg/crypto/keys/client/common.go:29]: TM2 shouldn't be aware of GNOHOME etc; let's pass the location from the caller, directly in gno.land/cmd/gnokey.

File: gno.land/pkg/integration/testing_integration.go [hunk 3]
```diff
@@ -51,7 +51,7 @@ func SetupGnolandTestScript(t *testing.T, txtarDir string) testscript.Params {
 
 	// `gnoRootDir` should point to the local location of the gno repository.
 	// It serves as the gno equivalent of GOROOT.
-	gnoRootDir := gnoland.GuessGnoRootDir()
+	gnoRootDir := gnoroot.MustGuessGnoRootDir()
```
Comment (moul) [inline gno.land/pkg/integration/testing_integration.go:54]:
```suggestion
	gnoRootDir := gnoroot.RootDir()
```

File: gno.land/pkg/integration/testing_integration.go [hunk 4]
```diff
@@ -12,7 +12,7 @@ import (
 	"testing"
 	"time"
 
-	"github.com/gnolang/gno/gno.land/pkg/gnoland"
+	"github.com/gnolang/gno/gnovm/pkg/gnoroot"
```
Comment (moul) [inline gno.land/pkg/integration/testing_integration.go:15]:
```suggestion
	"github.com/gnolang/gno/gnovm/pkg/gnoenv"
```

Comment (moul) [reply gno.land/pkg/integration/testing_integration.go:15]: and later, we'll move in gno.land/pkg, but right now we have a diamond dependency issue; so let's keep it in gnovm for now

File: gnovm/Makefile [hunk 5]
```diff
@@ -1,17 +1,24 @@
+GNOROOT_DIR ?= $(abspath $(lastword $(MAKEFILE_LIST))/../../)
+
+LDFLAGS := -X github.com/gnolang/gno/gnovm/pkg/gnoroot._GNOROOT=$(GNOROOT_DIR)
```
Comment (moul) [inline gnovm/Makefile:3]: move this just above GOBUILD_FLAGS, please

File: tm2/pkg/crypto/keys/client/common.go [hunk 6]
```diff
@@ -25,11 +25,19 @@ var DefaultBaseOptions = BaseOptions{
 func HomeDir() string {
```
Comment (moul) [inline tm2/pkg/crypto/keys/client/common.go:25]: move this helper to `pkg/gnoenv` with `GNOROOT`.

**Comment by thehowl** [comment]
realised we duplicated efforts on root dir :smiling_face_with_tear: in this pr: https://github.com/gnolang/gno/pull/1299

**Review by ajnavarro — APPROVED** [review]

**Pull request merged**